bundles/org.openhab.binding.kvv/src/main/java/org/openhab/binding/kvv/internal/KVVStopHandler.java

```diff
@@ -0,0 +1,177 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.kvv.internal;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.TimerTask;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.builder.ChannelBuilder;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+
+/**
+ * KVVStopHandler represents a stop and holds information about the trains
+ * which will arrive soon.
+ *
+ * @author Maximilian Hess - Initial contribution
+ */
+@NonNullByDefault
+public class KVVStopHandler extends BaseThingHandler {
+
+    @Nullable
+    private ScheduledFuture<?> pollingJob;
+
+    @Nullable
+    private KVVBridgeHandler bridgeHandler;
+
+    private KVVStopConfig config;
+
+    private boolean wasOffline;
+
+    public KVVStopHandler(final Thing thing) {
+        super(thing);
+        this.config = new KVVStopConfig();
+        this.wasOffline = false;
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        init(true);
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    private synchronized void init(final boolean createChannels) {
+        this.config = getConfigAs(KVVStopConfig.class);
+        if (config.stopId == "") {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Failed to get stop configuration");
+            return;
+        }
+
+        final Bridge bridge = getBridge();
+        if (bridge == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, "Failed to get bridge");
+            return;
+        }
+
+        final KVVBridgeHandler bridgeHandler = (KVVBridgeHandler) bridge.getHandler();
+        if (bridgeHandler == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_MISSING_ERROR, "Failed to get bridge handler");
+            return;
+        }
+
+        // creating channels
+        if (createChannels) {
+            final List<Channel> channels = new ArrayList<Channel>();
+            for (int i = 0; i < bridgeHandler.getBridgeConfig().maxTrains; i++) {
+                channels.add(ChannelBuilder.create(new ChannelUID(this.thing.getUID(), "train" + i + "-name"), "String")
```

Comment:
I tried to add channel groups during the last days without success: https://github.com/openhab/openhab-addons/pull/6484/commits/f388658dda2fdcadddaac628c90f4921c4e76605
Is there documentation or at least a minimal example?

Reply:
There is no complete manual on how to implement this. But because in this binding channel group and channel group types will be the same for all channels (only channels names and ids will be different per group/channel) you can define the channel group type, channel s in the group and channel types in xml, see https://www.openhab.org/docs/developer/bindings/thing-xml.html#channel-groups. You don't need the type providers here, you can just create the channels here by refering to the channel (group) type id as set in the xml definition. For example the networkupstools binding does this. (that binding also creates custom channel types, for specific quantity type channels, but that complexity is not needed here). If you have any additional questions don't hesitate to ask.

Reply:
Thanks a lot for your explanation!

I finally managed to introduce a set of channel types, but I am still not sure how this all fits together.
Is the definition of the channel types correct?
How do the channel types refer to group types?